### ISSUE TITLE
"undefined method" error with mime-types 3.1 / rest-client 1.6.7

### ISSUE THREAD

**Issue by henrik** [jage/elk#18]
Upgraded mime-types from 2.6.2 to 3.1.

This caused rest-client to downgrade from 1.8.0 to 1.6.7 to satisfy some gem's dependencies.

When calling `Elk::SMS.send(…)` I then got errors like:

> NoMethodError: undefined method `[]' for #<Set: {#<MIME::Type: application/json>}>

So maybe there's an incompatibility here and the Elk gem could either lock down its version dependencies further (as a quick fix) or support more versions of mime-types (as a longer-term fix).

**Comment by henrik** [comment]
@46elks @littlekid Seems this gem might be unmaintained. Don't suppose you want to adopt it, if @jage would be OK with that? :)

**Comment by jage** [comment]
> if @jage would be OK with that? :)

Sure! 

Yeah unfortunately I don't have much time for this gem, sorry!
I'll see if I might be able to fix #17 and #16 which hopefully would make it easier to improve the gem's dependency compatibility.

**Comment by jage** [comment]
@henrik could you paste your `Gemfile` and `Gemfile.lock` so I can try out your combination of gems?

Also, if you have time maybe you could try out master (7b413a33a56d0a2116e6e570122371371a5eb13a) and see if it works better with rest-client 2.0 which should allow newer mime-types.

**Comment by henrik** [comment]
Hi @jage, thank you!

I'm uncomfortable sharing the whole of those files for this project, but I tried it again and looked closer, and what seems to be happening with mime-types/rest-client is basically this:

``` diff
-    rest-client (1.8.0)
-      http-cookie (>= 1.0.2, < 2.0)
-      mime-types (>= 1.16, < 3.0)
-      netrc (~> 0.7)
+    rest-client (1.6.7)
+      mime-types (>= 1.16)
```

When I update mime-types to 3.1, then rest-client 1.8.0 is no longer compatible with it (only supports mime-types "< 3.0"), so Bundler finds another rest-client (1.6.7) which does not exclude mime-types 3.1.

I'm trying out master now – I've been able to update all the libs and things seem fine in dev, but I'll keep you posted if CI or production feel differently. Thanks again! :)
